server_test.go

```diff
@@ -0,0 +1,77 @@
+package osquery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/kolide/osquery-go/gen/osquery"
+	"github.com/kolide/osquery-go/plugin/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockExtensionClient struct {
+	mut    sync.Mutex
```

Comment:
AFAICT this mutex is unnecessary to repro the issue. I think we just need to have `RegisterExtension` return an error, and that will be sufficient to get the hang. Am I missing something?